### DISCUSSION TITLE
Automatically assign budget to transaction based on the Tags

### DISCUSSION THREAD

**Discussion by M1scer** [firefly-iii/firefly-iii#7885] [Firefly III] [answered]
I would like to assign tags to my transactions and automatically set the corresponding budget based on these tags.

I created a rule that checks transactions for the tags when they are changed and then adds them to the budget.
I have also created a rule that automatically adds tags to some transactions.
Like this:
New Transaction -> Rule adds a Tag -> Rule adds Budet based on Tag.

If I now run the rule manually to add the tags, the second rule should actually add these transactions to a budget afterwards.
However, this does not happen.

If I check the second rule (assign transactions to budgets based on keywords), i.e. see which transactions would affect this rule, then I see the corresponding ones. So the rule seems to recognize the transactions but not trigger them.

Does adding a tag even count as a change to a transaction?

Edit: Not yet tested, but i guess it would be the same with Categories instead of Budgets.

Edit 2: If a tag is added manually, the second rule takes effect as it should.
I guess the "Submit Options" -> "Apply Rules" which can be selected when adding manually is not triggered when adding automatically.

**Comment by JC5** [comment]
> Does adding a tag even count as a change to a transaction?

Not from a rule, no. That could lead to eternal loops. I will see if I can replicate this issue.

**Reply by M1scer** [reply]
The loops can of course be a problem. 
Maybe a checkbox in the rule that activates further rules with the hint that this can trigger a loop?

**Comment by M1scer** [comment, marked as answer]
I found a way to solve my problem.
With the command line interface i will apply all rules via a cronjob. Then at least after a running the cronjob, the correct budgets and categories are applied.

My command looks like this:
`docker exec -it fireflyApp php artisan firefly-iii:apply-rules --user=1 --rule_groups=4,5 --accounts=300,301,302,303 --token=xxxxxxxx`

However, I have noticed that only rules that have "new transaction created" as a trigger are taken into effect. If the trigger is set to "When a transaction is update", they are ignored.

I think all rules should be taken into effect?

~~In my case, I have to duplicate all the rules that actually have the trigger "When a transaction is updated" to be taken into account in the cronjob as well as in an actual update.~~
Edit: i dont need to copy rules, just wait for the next cronjob.
